### ISSUE TITLE
Ghost stack

### ISSUE THREAD

**Issue by Faexa** [photoprism/photoprism#3526]
Hello,
I've encountered a bug with the picture stacking.

I don't really know what append in the first place. I've this layout for my pictures : /2022/2022.09 My event/Subfolder/xx.jpg

I initially put all my photos a month ago (you will see the differences in index times below) in the same folder /2022/2022.09 My event/.  Changed my mind, I decided to edit the metadata with Digikam, move the pictures to a temp folder  beginning with __ to hide them from Photoprism, launch a full rescan which made these photo disappear. I did the editing and sorted the pics into subfolders.  (the metadata editing will explain the hash differences in the database)
After that, I did a full rescan.

Everything is fine, but all of these pictures got listed into  "Stacks".
When clicking each picture to see the details, it showed only one file in the stack, so no possibility to unstack...

Looking in the database, table files, I found that it stacked with missing files entry, like this : 

| id | photo_id | photo_taken_at | photo_uid | time_index | media_id | file_uid | file_name | file_hash | file_missing | created_at | updated_at |  deleted_at |
| --- | --- | --- | --- | --- | --- | --- | --- | --- | -- | -- | -- | -- |
| 1248 | 1221 |  2022-09-04 00:36:51 | prwtfxu3pjgga7un | NULL | NULL | frwtfxu3iu05892i | 2022/2022.09 Event/Photos Chris/event_time_1234.jpg | 18ad82042091a21a2657a7cf63ee06ea3c718f7e | 1 | 2023-06-25 15:49:06 | 2023-06-25 15:49:06 | 2023-06-27 11:06:53 |
| 1963 | 1221 | 2022-09-04 00:36:51 | prwtfxu3pjgga7un | 79779095996349-9999998779-0-frxdgwo3s8cjxkgc | 9999998779-0-frxdgwo3s8cjxkgc | frwtfxu3iu05892i | 2022/2022.09 Event/Photos Chris/01 Subfolder/event_time_1234.jpg | afbd7fa4eb9dd2d1df2f7cbd10218d6d5df38876 | 0 | 2023-07-06 11:22:00 | 2023-07-06 17:37:53 | NULL |

Removing the duplicate solved the problem.

It's a very very unlikely bug. 
But I would propose maybe to not show a file as stacked if "file_missing" = 1 or "deleted_at" not NULL ?

Alternatively, is it possible to clean all these missing/deleted files entries with CLI ?

Thank you very much for your work and your time

**Comment by lastzero** [comment]
Did you already try the photoprism purge and cleanup CLI commands?

**Comment by Faexa** [comment]
Did this one without success :
docker compose exec photoprism photoprism index --cleanup

Didn't knew about purge

I since deleted from the database all entries with deleted_at not NULL. I tried to reproduce the initial bug moving around a photo, no success. It updated the file entry with missing and delated_at, but didn't stack with another one when I reintroduce the picture at another location. And cleanup removed this entry.

I'll see if it happen again and will try cleanup/purge
